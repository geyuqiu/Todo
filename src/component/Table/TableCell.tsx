import React from "react";
import {styled} from "twin.macro";
import cn from 'classnames';

import {getStyles} from "./TableCell.styles";

type TableCellProps = {
	variant?: "start" | "middle" | "end";
	className?: string;
	innerClassName?: string;
	isCompact?: boolean;
	children: React.ReactNode;
} & React.HTMLProps<any>;

const TableCellInnerWrapper = styled.div<TableCellProps>(getStyles);

export const TableCell = ({variant, className, innerClassName, isCompact, children, ...props}: TableCellProps) => (
	<td className={cn('border-gray-lightest border', className)} {...props}>
		<TableCellInnerWrapper variant={variant} className={innerClassName} isCompact={isCompact}>
			{children}
		</TableCellInnerWrapper>
	</td>
);

TableCell.defaultProps = {
	variant: "middle",
};
