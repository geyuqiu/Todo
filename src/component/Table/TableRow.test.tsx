import {render} from "@testing-library/react";
import React from "react";

import {TableRow} from "./TableRow";
import {TableCell} from './TableCell';

describe("TableRow", () => {
	it("should render with border", () => {
		const {asFragment} = render(
			<table>
				<tbody>
					<TableRow border={true}>
						<TableCell>
							something
						</TableCell>
					</TableRow>
				</tbody>
			</table>,
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it("should render without border", () => {
		const {asFragment} = render(
			<table>
				<tbody>
					<TableRow border={false} >
						<TableCell>
							something
						</TableCell>
					</TableRow>
				</tbody>
			</table>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
