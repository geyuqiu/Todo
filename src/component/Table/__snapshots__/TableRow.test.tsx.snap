// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableRow should render with border 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-990v33-0 iJGVRV"
        data-testid="TableRow"
      >
        <td
          class="border-gray-lightest border"
        >
          <div
            class="TableCell__TableCellInnerWrapper-l6uokn-0 BHpwz"
          >
            something
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`TableRow should render without border 1`] = `
<div>
  <table>
    <tbody>
      <tr
        class="TableRow__TableRowStyled-sc-990v33-0 cGVoQL"
        data-testid="TableRow"
      >
        <td
          class="border-gray-lightest border"
        >
          <div
            class="TableCell__TableCellInnerWrapper-l6uokn-0 BHpwz"
          >
            something
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
